src/CustomElementRegistry.js

```diff
@@ -101,7 +100,7 @@ export default class CustomElementRegistry {
     this._internals.setDefinition(localName, definition);
 
     if (this._upgradeOnDefine) {
-      this._internals.upgradeTree(document);
+      this._internals.patchAndUpgradeTree(document);
```

Comment:
See https://github.com/webcomponents/custom-elements/pull/43/files/d3c10972dee961702ab5248dc871416207d339c0..2f22e61d91e1f2ef8be6c5bbb53d6a702ecb819b#r97923523; this is the work that needs to be gated on a debounced _call_ to `documentReady` (rather than checking a flag). The function call as opposed to a flag is important so that imports polyfill can synchronously check for newly added imports.